shop/forms/checkout.py

```diff
@@ -168,9 +168,12 @@ def form_factory(cls, request, data, cart):
                 # prevent adding the same address twice
                 all_field_names = [f.name for f in cls.get_model()._meta.get_fields()]
                 filter_args = dict((attr, val) for attr, val in address_form.data.items()
-                                   if attr in all_field_names and val)
+                                   if attr in all_field_names)
```

Comment:
This change is more or less unrelated. If a field is empty in one address but not in another, we should treat these as two different addresses.